### PR TITLE
Pre-prompt for Latex

### PR DESCRIPTION
**Pull request by PopDaph**
## Description

Issue: https://dustcommunity.slack.com/archives/C06SHT0F20G/p1729083880269249
Repro of the issue: https://dust.tt/w/0ec9852c2f/assistant/3zBCIyWOuw

Adding a pre-prompt for Latex to make sure the model generates it properly. 
I know this is not a scalable solution to add some pre-prompt for everything but at the same time this is the most logic fix (I would not be chocked if we don't want to merge this though)

## Risk

/

## Deploy Plan

Deploy front.

### CODE REVIEW

File: front/lib/api/assistant/generation.ts [hunk 1]
```diff
@@ -468,6 +468,9 @@ export async function constructPromptMultiActions(
     }
   }
 
+  additionalInstructions +=
+    "If asked to generate a formula in latex, put the formula between $$.";
```
Comment (spolu) [inline front/lib/api/assistant/generation.ts:472]: You're missing some `\n` I think? 

"If asked to generate a formula" > "When generating latex math formulas, rely on the $$ escape sequence, single $ latex sequence are not supported" ?

Comment (PopDaph) [reply front/lib/api/assistant/generation.ts]: Much better, applied 👍🏻

File: front/lib/api/assistant/generation.ts [hunk 2]
```diff
@@ -468,6 +468,9 @@ export async function constructPromptMultiActions(
     }
   }
 
+  additionalInstructions +=
+    "When generating latex formulas, solely rely on the $$ escape sequence, single $ latex sequence are not supported.\n";
```
Comment (spolu) [inline front/lib/api/assistant/generation.ts:472]: I would put one `\n` before as well out of consistency with the above but no strong feeling

**Review by spolu — APPROVED** [review]
LGTM

File: front/lib/api/assistant/generation.ts (hunk 2 above)
Comment (spolu) [reply front/lib/api/assistant/generation.ts:472]: Also my bad but I believe it should be

"single $ latex sequenceS" but to be double checked

Comment (PopDaph) [reply front/lib/api/assistant/generation.ts]: Some have only an ending `\n`, some have both indeed.
 I added both to mark it's unrelated to the previous instruction!
I also fixed the typo, thanks 🙏🏻

**Pull request merged**